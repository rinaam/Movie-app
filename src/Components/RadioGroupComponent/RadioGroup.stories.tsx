import React from "react";
import { Story, Meta } from "@storybook/react";

import RadioGroupComponent, { IRadioGroupProps } from "./RadioGroupComponent";

export default {
  title: "RadioButton",
  component: RadioGroupComponent,
} as Meta;

const Template: Story<IRadioGroupProps> = (args) => (
  <RadioGroupComponent {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  items: [
    {
      name: "option1",
      id: 1,
    },
    {
      name: "option2",
      id: 2,
    },
    {
      name: "option3",
      id: 3,
    },
  ],
  groupName: "fake",
  title: "Select:",
  onChange: () => {
    // do nothing
  },
};
