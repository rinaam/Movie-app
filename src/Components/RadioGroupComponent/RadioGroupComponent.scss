@use "sass:map";
@import "../../Styles/variables.scss";

.radio {
  display: flex;
  flex-direction: column;
  &-title {
    margin: map.get($spacing, "m") 0px;
    font: {
      size: map.get($font-size, "m");
      weight: map.get($font-weight, "medium");
    }
  }
  &-option {
    margin: map.get($spacing, "s") 0px;
    display: flex;
    flex-direction: row-reverse;
    justify-content: flex-end;
    &-btn {
      cursor: pointer;
    }
  }
}
