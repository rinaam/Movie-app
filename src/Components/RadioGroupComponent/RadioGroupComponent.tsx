import React from "react";
import "./RadioGroupComponent.scss";

export interface IRadioGroupProps {
  items: TItem[];
  groupName: string;
  title?: string;
  onChange: (event: string) => void;
}

export type TItem = {
  name: string;
  id: number;
};

const RadioGroupComponent: React.FC<IRadioGroupProps> = ({
  items,
  groupName,
  title,
  onChange,
}) => (
  <div className="radio">
    <p className="radio-title">{title}</p>
    {items.map((item) => (
      <label className="radio-option" key={item.id}>
        {item.name}
        <input
          type="radio"
          name={groupName}
          value={item.name}
          onChange={() => onChange(item.name)}
          className="radio-option-btn"
        />
      </label>
    ))}
  </div>
);

export default RadioGroupComponent;
