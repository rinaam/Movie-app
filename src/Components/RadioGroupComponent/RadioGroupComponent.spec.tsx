import { create } from "react-test-renderer";
import RadioGroupComponent from "./RadioGroupComponent";
import React from "react";

describe("RadioGroupComponent", () => {
  it("should render RadioGroupComponent", () => {
    const fakeChange = jest.fn();
    const fakeItems = [
      {
        name: "option1",
        id: 1,
      },
      {
        name: "option2",
        id: 2,
      },
      {
        name: "option3",
        id: 3,
      },
    ];
    const wrapper = create(
      <RadioGroupComponent
        items={fakeItems}
        groupName="fakeName"
        onChange={fakeChange}
      />
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger on change", () => {
    const fakeChange = jest.fn();
    const fakeItems = [
      {
        name: "option1",
        id: 1,
      },
      {
        name: "option2",
        id: 2,
      },
      {
        name: "option3",
        id: 3,
      },
    ];
    const wrapper = create(
      <RadioGroupComponent
        items={fakeItems}
        groupName="fakeName"
        onChange={fakeChange}
      />
    );
    const option2 = wrapper.root.findByProps({ value: "option2" });

    option2.props.onChange();

    expect(fakeChange).toBeCalled();
  });
});
