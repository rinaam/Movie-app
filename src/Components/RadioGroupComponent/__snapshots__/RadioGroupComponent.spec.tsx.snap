// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroupComponent should render RadioGroupComponent 1`] = `
<div
  className="radio"
>
  <p
    className="radio-title"
  />
  <label
    className="radio-option"
  >
    option1
    <input
      className="radio-option-btn"
      name="fakeName"
      onChange={[Function]}
      type="radio"
      value="option1"
    />
  </label>
  <label
    className="radio-option"
  >
    option2
    <input
      className="radio-option-btn"
      name="fakeName"
      onChange={[Function]}
      type="radio"
      value="option2"
    />
  </label>
  <label
    className="radio-option"
  >
    option3
    <input
      className="radio-option-btn"
      name="fakeName"
      onChange={[Function]}
      type="radio"
      value="option3"
    />
  </label>
</div>
`;
